src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsAmazonObject.java

```diff
@@ -44,17 +54,28 @@ public void verifyCallToAmazonIssued() {
         }
     }
 
-    public void verifyAdFromAmazonPresent() {
-        waitForElementByElement(slotWithAmazon);
-        WebElement amazonIframe = slotWithAmazon.findElement(By.cssSelector(
-                "div[id*=__container__] > iframe"
-        ));
-        driver.switchTo().frame(amazonIframe);
+    public AdsAmazonObject verifyAdFromAmazonPresent() {
+        driver.switchTo().frame(getAmazonIframe(slotWithAmazon));
+
         if (checkIfElementOnPage(AMAZON_IFRAME)) {
             PageObjectLogging.log("AmazonAd", "Script returned by Amazon present", true);
         } else {
             throw new NoSuchElementException("Amazon Ad not found on page");
         }
+
+        driver.switchTo().defaultContent();
+        return this;
+    }
+
+    public void verifyNoAdsFromAmazonPresent() {
```

Comment:
IMO The method should be consistent with `verifyAdFromAmazonPresent`.
In one method we throw exception if we got a negative result, in another one - logging things.
I would rewrite both method to:
```java
  public AdsAmazonObject verifyAdFromAmazonPresent() {
    driver.switchTo().frame(getAmazonIframe(slotWithAmazon));
    Assertion.assertTrue(checkIfElementOnPage(AMAZON_IFRAME));
    PageObjectLogging.log("AmazonAd", "Script returned by Amazon present", true);
    driver.switchTo().defaultContent();
    return this;
  }
```
and
```java
  public AdsAmazonObject verifyNoAdsFromAmazonPresent() {
    driver.switchTo().frame(getAmazonIframe(slotWithAmazon));
    Assertion.assertFalse(checkIfElementOnPage(AMAZON_IFRAME));
    PageObjectLogging.log("AmazonAd", "No Amazon ad present", true);
    driver.switchTo().defaultContent();
    return this;
  }
```